### PR TITLE
Copy hotkeys

### PR DESCRIPTION
**Pull request by Harjot1Singh**
To do:
- [x] "lift" functionality out of Display component, and move shortcut into another (`Presenter/index.js` most likely)
- [x] Reset Settings does not work (simulate a crash). Server overwrites this, since server passes along changes, not removals. Have to signal for remove somehow, have a separate event for removal, or flip-over to passing the current user's settings, always
- [x] Reset/remove settings. Related to above.
- [ ] Configure hotkeys

### CODE REVIEW

File: app/frontend/src/shared/CopyHotkeys.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  const hotkeyHandlers = !!line && [
+    [ COPY_SHORTCUTS.copyGurmukhi.name, line.gurmukhi ],
+    [ COPY_SHORTCUTS.copyEnglishTranslation.name, translations.english ],
+    [ COPY_SHORTCUTS.copyPunjabiTranslation.name, translations.punjabi ],
+    [ COPY_SHORTCUTS.copySpanishTranslation.name, translations.spanish ],
+    [ COPY_SHORTCUTS.copyEnglishTransliteration.name, transliterations.english ],
+    [ COPY_SHORTCUTS.copyHindiTransliteration.name, transliterations.hindi ],
+    [ COPY_SHORTCUTS.copyUrduTransliteration.name, transliterations.urdu ],
+    [ COPY_SHORTCUTS.copyAuthor.name, getAuthor() ],
+  ].reduce( ( hotkeys, [ name, content ] ) => ( {
```
Comment (AkalUstat) [inline app/frontend/src/shared/CopyHotkeys.js]: Wooaaah thats hecka cool AND clean! How does this work?

Comment (Harjot1Singh) [reply app/frontend/src/shared/CopyHotkeys.js]: Take a look at the `lib/hooks.js`, those are some re-usable state hooks that make it easy to just retrieve whatever data I need.

The reducer then takes all of the hotkey names and content mappings, and registers the `copyToClipboard` handler with it.

See https://github.com/ShabadOS/desktop/pull/400/commits/7f2ee3562eb24c2294dfb703cb4b5d4a0e8ae3fd to see it in isolation

File: app/frontend/src/App.js [hunk 2]
```diff
@@ -151,34 +168,36 @@ class App extends PureComponent {
...
+      [ BookmarksContext.Provider, banis ],
+      [ WritersContext.Provider, writers ],
+      [ RecommendedSourcesContext.Provider, recommendedSources ],
+      [ ContentContext.Provider, { bani, shabad, lineId, mainLineId, nextLineId } ],
+      [ SnackbarProvider ],
+    ].reduce( ( withContexts, [ Provider, value ] ) => children => withContexts(
+      <Provider value={value}>
+        {children}
+      </Provider>,
+    ), context => context )
```
Comment (AkalUstat) [inline app/frontend/src/App.js]: :eyes: soo smart brooo idk where you get all these ideas from

**Comment by Harjot1Singh** [comment]
Moving configuring hotkeys out of this PR

**Pull request merged**